src/map/ai/helpers/targetfind.cpp

```diff
@@ -243,12 +243,27 @@ void CTargetFind::addAllInAlliance(CBattleEntity* PTarget, bool withPet)
 
 void CTargetFind::addAllInParty(CBattleEntity* PTarget, bool withPet)
 {
-
     PTarget->ForParty([this, withPet](CBattleEntity* PMember)
     {
         addEntity(PMember, withPet);
+        if (PMember->objtype == TYPE_PC) {
+            CCharEntity* PChar = (CCharEntity*)m_PBattleEntity;
```

Comment:
this doesn't seem right - first of all, there's no guarantee that m_PBattleEntity is a CharEntity (you checked PMember in the previous line, not m_PBattleEntity).  Then in the next line, check if PChar (which was from m_PBattleEntity) is a TRUST.. which is it? a player or a trust?
This whole block before `else if (PChar->PTrusts.size() != 0) {` I think should just be removed.  AFAIK, the only thing you want to do is add the trusts if the target is a player that has trusts.  This block is trying to add trusts if it found a trust in the loop? But if it found a trust in the loop, it wouldn't have to find the master and add the trusts because it already found one (evidenced by the fact that it is currently iterating on a trust)

Reply:
addAllInParty doesn't see trusts as members.  I'll be resubmiting the new code for this shortly.  It looks to see if the entity is a trust, and then to get its master and iterate thru the masters Trust list to add the effect to the other trusts.  I also added a modelID check for now so it won't cast the effect on the casting trust twice.